runtime/cudaq/builder/kernels.h

```diff
@@ -8,111 +8,122 @@
 
 #pragma once
 
+#include "fmt/format.h"
```

Comment:
I hate to say it because I like using it, but it would be best to keep this fmt header out of a public header that a programmer may include. It would force us to keep fmt around in the deployment and auto include its headers with nvq++.